apps/happy-blocks/src/pricing-plans/hooks/pricing-plans.ts

```diff
@@ -0,0 +1,207 @@
+import formatCurrency from '@automattic/format-currency';
+import { useEffect, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import config from '../config';
+import { ApiPricingPlan, PricingPlansConfiguration } from '../types.js';
+
+const DAYS_IN_MONTH = 31;
+const MONTHS_IN_YEAR = 12;
+
+/**
+ * Returns a formatted motnhly price for a plan.
+ */
+const getMonthlyPrice = ( plan: ApiPricingPlan ): string | null => {
+	return formatCurrency(
+		Math.round( ( plan.raw_price / plan.bill_period ) * DAYS_IN_MONTH ),
+		plan.currency_code,
+		{
+			stripZeros: true,
+		}
+	);
+};
+
+/**
+ * Returns a description of the savings of the yearly plan over the monthly plan.
+ */
+const getSavingsDescription = (
+	yearlyPlan: ApiPricingPlan,
+	monthyPlan: ApiPricingPlan
+): string => {
+	const yearlyPricePerMonth = yearlyPlan.raw_price / MONTHS_IN_YEAR;
+	const savings = Math.round( 100 - ( yearlyPricePerMonth / monthyPlan.raw_price ) * 100 );
+
+	if ( savings <= 0 ) {
+		return '';
+	}
+
+	// eslint-disable-next-line @wordpress/valid-sprintf
+	return sprintf(
+		// translators: %s is a percentage, e.g. "16%"
+		__( 'save %s%%', 'happy-blocks' ),
+		savings
+	);
+};
+
+/**
+ * Parse the API response into a more usable format.
+ */
+const parsePlans = ( data: ApiPricingPlan[] ): PricingPlansConfiguration | null => {
+	const premiumYearly = data.find( ( plan ) => plan.path_slug === 'premium' );
```

Comment:
Nit: Probably premature optimization but what do you think about doing one loop over `data` and building a better plans structure out of it instead of all the `.find`s?